File: gs-spark/gs-spark-java/src/main/java/com/tirthal/learning/util/SparkUtils.java
package com.tirthal.learning.util;

import org.apache.spark.sql.SparkSession;

public class SparkUtils {

    public static SparkSession getSparkSession(String appName) {

        // Run Spark locally with one worker thread (i.e. no parallelism at all)
        return SparkSession
                .builder()
                .appName(appName)
                .config("spark.master", "local") // change to "local[*]" for parallelism
                .getOrCreate();
    }
}

File: gs-spark/gs-spark-scala/src/main/scala/com/tirthal/learning/util/SparkUtils.scala
package com.tirthal.learning.util

import org.apache.spark.sql.SparkSession

object SparkUtils {

  // Run Spark locally with one worker thread (i.e. no parallelism at all)
  def getSparkSession(appName: String): SparkSession = {
    SparkSession
      .builder
      .appName(appName)
      .config("spark.master", "local")  // change to "local[*]" for parallelism
      .getOrCreate
  }
}
